lib/matplotlib/tests/test_rcparams.py

```diff
@@ -342,6 +342,8 @@ def generate_validator_testcases(valid):
                      ('', ''), (' ', ' '),
                      ('None', 'none'), ('none', 'none'),
                      ('DoTtEd', 'dotted'),
+                     (b'dotted', 'dotted'),  # binary string
```

Comment:
These should only pass on python2 (where str and bytes are conflated).  In python 3 this is correctly failing as`b'dotted' != 'dotted'` because to do the comparison you would have to either decode the bytes (which we can't do because we don't know the encoding) or encode the string into the same encoding as the bytes (which we can't do because we don't know the encoding). 

I suggest putting this in another test with a skip if python3 mark?